src/components/game/ScreenshotPage.js

```diff
@@ -159,6 +160,26 @@ export default function ScreenshotPage({
         )
       })
   }
+  const handleReady = async (event) => {
+    event.preventDefault()
+    try {
+      if (isReady) {
+        globalContext.socket.emit('notRoundReady', globalContext.socket.id)
+      } else {
+        console.log('emitting roundReady')
+        globalContext.socket.emit('roundReady', globalContext.socket.id)
+      }
+
+      await new Promise((resolve) => {
+        globalContext.socket.once('roomUsers', async (data) => {
+          resolve(data)
+        })
+      })
```

Comment:
What does this function do?

Reply:
This function handles the 'I'm Ready' button on the Screenshot Page. It is called when the "I'm Ready' button is clicked on, and it emits whether the player is ready to start the next round or not.

Reply:
Sorry I wasn't clear I just meant the function in the socket.once(). It looks like it just waits to receive data then doesn't really do anything with the data upon receiving it unless I'm missing something. IK we do something similar elsewhere but idk if we need it here.

Reply:
nvm I get it know, we wait for the server to respond before we update the local variable to make sure the server and client are on the same page in case there was some interruption,